packages/editor-ui/src/utils/nodeViewUtils.ts

```diff
@@ -1127,17 +1143,20 @@ export const getJSPlumbConnection = (
 	const sourceEndpoint = getOutputEndpointUUID(sourceId, connectionType, sourceOutputIndex);
 	const targetEndpoint = getInputEndpointUUID(targetId, connectionType, targetInputIndex);
 
-	const sourceNodeOutput = sourceNodeType?.outputs?.[sourceOutputIndex] || NodeConnectionType.Main;
+	const sourceNodeOutput = sourceNodeType?.outputs?.[sourceOutputIndex] ?? NodeConnectionType.Main;
 	const sourceNodeOutputName =
-		typeof sourceNodeOutput === 'string' ? sourceNodeOutput : sourceNodeOutput.name;
+		typeof sourceNodeOutput === 'string'
+			? sourceNodeOutput
+			: 'name' in sourceNodeOutput
+				? `${sourceNodeOutput.name}`
+				: '';
 	const scope = getEndpointScope(sourceNodeOutputName);
 
-	// @ts-ignore
 	const connections = instance?.getConnections({
 		scope,
 		source: sourceId,
 		target: targetId,
-	}) as Connection[];
+	} as SelectOptions<Element>) as Connection[];
 
 	return connections.find((connection: Connection) => {
```

Comment:
```suggestion
	} as SelectOptions<Element>);
        
        if (!Array.isArray(connections)) return;
        
	return connections.find((connection: Connection) => {
```
We can return early if not an array, since this would fail anyway

Reply:
Great idea, updated!